### PR TITLE
Map preview on cart

### PR DESCRIPTION
**Pull request by Lenninlasd**
## What does this PR do?

Map preview everything on the cart, not only the selected item.
### Screenshot

iy8lcyolj3

### CODE REVIEW

File: src/common/addlayers/RegistryLayersDirective.js [hunk 1]
```diff
@@ -183,9 +183,14 @@
               }
             }
 
+            scope.toggleCart = function(layerConfig) {
```
Comment (milafrerichs) [inline src/common/addlayers/RegistryLayersDirective.js]: can you add tests for the new scope methods?

File: src/common/addlayers/RegistryLayersDirective.js [hunk 2]
```diff
@@ -224,6 +229,10 @@
               ];
             };
 
+            scope.cleanMapPreview = function() {
```
Comment (milafrerichs) [inline src/common/addlayers/RegistryLayersDirective.js:232]: can you add tests for the new scope methods?

File: src/common/addlayers/RegistryLayersDirective.js [hunk 3]
```diff
@@ -269,6 +278,15 @@
 
             // load saved registry layers if they exist
             scope.addRegistryLayersFromSavedMap();
+
+            function cartLayer(layerConfig) {
```
Comment (milafrerichs) [inline src/common/addlayers/RegistryLayersDirective.js]: `cartLayer` is not a very good name for this method. 
Does it reset the previewLayers to the layerConfig that gets passed into it, correct?
Then a name might be `resetPreviewLayersWithConfig`

File: src/common/addlayers/RegistryLayersDirective.js (hunk 1 above)
Comment (milafrerichs) [inline src/common/addlayers/RegistryLayersDirective.js]: Is `toggleCart` a good name for this method?
It does more than just to "toggle" the layer, right?
Why does it reset the previewLayers to just the one that is toggled? Shouldn't all layers be visible that are in the cart?

File: src/common/map/MapDirective.js [hunk 4]
```diff
@@ -53,15 +53,44 @@
 
             scope.$watch('layers', function(layers) {
               if (layers && map) {
-                var layerLength = map.getLayers().getLength();
-                for (var i = 1; i < layerLength; i++) {
-                  map.removeLayer(map.getLayers().getArray()[1]);
+                deleteNoCartLayers();
```
Comment (milafrerichs) [inline src/common/map/MapDirective.js]: why do you remove all layers and then in addAllLayers you remove them once again?

File: src/common/map/MapDirective.js [hunk 5]
```diff
@@ -53,15 +53,44 @@
...
+                addAlllayers(layers);
+              }
+            });
+
+            function deleteNoCartLayers() {
```
Comment (milafrerichs) [inline src/common/map/MapDirective.js]: there is nothing with the cart involved in this method. why does it have cart in the name?

File: src/common/addlayers/RegistryLayersDirective.js (hunk 3 above)
Comment (Lenninlasd) [reply src/common/addlayers/RegistryLayersDirective.js]: @milafrerichs this function just add the to the map preview the layers that are inside the cart.

File: src/common/map/MapDirective.js (hunk 4 above)
Comment (Lenninlasd) [reply src/common/map/MapDirective.js]: First I delete the layers that were added in the map preview making hover through  `deleteNoCartLayers` function. 
In `addAllLayers` I delete only a repeated layer  with layerId property.

**Comment by ingenieroariel** [comment]
Hello @Lenninlasd @milafrerichs - what can we do with this changeset? I'd like to try it out but understand it may need more work before it can land.

File: src/common/map/MapDirective.js (hunk 4 above)
Comment (milafrerichs) [reply src/common/map/MapDirective.js]: can you name the function accordingly?

**Comment by milafrerichs** [comment]
I think @Lenninlasd is working at it right now and once he fixed a few of these we can merge but need to get back to it to finish it... 
it's your call...

**Comment by ingenieroariel** [comment]
@milafrerichs I'll let you click the merge button then, let's leave a `TODO` comment with the stuff that we did not get to today.

**Comment by Lenninlasd** [comment]
@ingenieroariel the test fail when is merged with production.
Merge first PR:  https://github.com/terranodo/MapLoom/pull/77

**Pull request merged**